index.html

```diff
@@ -2595,6 +2595,139 @@ <h3>Data Schemas</h3>
 
       </section>
 
+      <section>
+        <h2>External Resource Integrity</h2>
+        <p>
+          When including a link to an external resource in a VC, it is 
+          desirable to know whether the resource that is pointed to is the
+          same at signing time as at verification time.  This applies
+          to cases where there is an external resource that is
+          remotely retrieved as well as to cases where the issuer and/or
+          verifier may have local cached copies of a resource.
+        </p>
+        <p>
+          It is also desirable to know that the contents of the
+          context(s) used in the verifiable credential are the same when
+          used by both the issuer and verifier. 
+        </p>
+        <p>
+          To validate that a resource referenced by a Verifiable Credential is
+          the same at verification time as at issuing time, an implementer
+          MAY include a property named <code>resourceIntegrity</code> that
+          stores an array of objects that describe additional integrity
+          metadata about each resource referenced by the VC. If
+          <code>resourceIntegrity</code>
+          is present, there MUST be an object in the array for each remote
+          resource for each context used in the verifiable credential. 
+        </p>
+        <p>
+          Each object in the
+          <code>resourceIntegrity</code> array MUST contain the following:
+          the [[URL]] to the resource named <code>id</code> and the 
+          <code>integrity</code> information for the resource
+          constructed using the method specified in <a
+          href="https://www.w3.org/TR/SRI/#integrity-metadata">Subresource
+          Integrity</a>.
+        </p>
+        <p>
+          There MUST NOT be more than one object in the
+          <code>resourceIntegrity</code> per <code>id</code>. 
+        </p>
+        <p>
+          When performing integrity checking the
+          strongest hash algorithm available SHOULD be used to confirm
+          integrity of the resource.
+        </p>
+        <p>
+          An object in the <code>resourceIntegrity</code> array MAY
+          contain a property named <code>timestamp</code>
+          that indicates the time at which the hash was computed. The
+          <code>timestamp</code> property if included MUST be a string
+          value of an [[XMLSCHEMA11-2]] combined date-time string.
+        </p>
+        <p>
+          An object in the <code>resourceIntegrity</code> array MAY contain
+          a property named <code>mediaType</code> that indicates the 
+          expected media type for the indicated <code>resource</code>.
+          If a <code>mediaType</code> is included it must be a valid
+          media type as listed in the 
+          <a href="https://www.iana.org/assignments/media-types/media-types.xhtml">
+            IANA Media Types
+          </a> registry.
```

Comment:
```suggestion
          When using the <code>mediaType</code> property, implementers are advised to 
          only use values listed in the 
          <a href="https://www.iana.org/assignments/media-types/media-types.xhtml">
            IANA Media Types
          </a> registry.
```

This is not testable unless all implementations include the latest, complete media type table in their implementations, which is a burden that we shouldn't place on implementers.

Reply:
does SHOULD work for you here?

Reply:
I wouldn't object to a SHOULD, but do warn that it doesn't help us to use normative language around things that are fundamentally too difficult to test (because they're a moving target, like the IANA Media Type registry). 

The purpose of most normative language is to provide guidance to developers on what they should implement. In this case, I can't think of a developer that would seriously consider encoding the media type table in their software and rejecting/warn if someone were to use a non-listed media type. That's the reason I suggested "advised to" because that's what we're really trying to convey here. This isn't guidance on what an implementation should/shouldn't do -- it's guidance to implementers to do the right thing wrt. the input they provide. 

I also question the guidance, because it's arguable that unregistered media types (or vendor-based media types) are fine if both sender and receiver know how to process them (though, it does lead to interop issues). All this to say, this statement is not as straightforward as it seems. I'd prefer we say nothing or use the "advised to" language. Going further is suggesting that it's more enforceable than it is.